src/_langs/en/updates/2015-06-12-Media-Source-Extensions-for-Audio.markdown

```diff
@@ -0,0 +1,373 @@
+  // entire segment at once, but we could also retrieve it in chunks and append
+  // each chunk separately.  MSE will take care of assembling the pieces.
+  GET('sintel/sintel_0.mp3', function(data) { onAudioLoaded(data, 0); } );
+}, false);
+
+audio.src = window.URL.createObjectURL(mediaSource);
+{% endhighlight %}
+
+Once the MediaSource object is connected, it will perform some initialization and eventually fire a "sourceopen" event; at which point we can create a [SourceBuffer](http://www.w3.org/TR/media-source/#sourcebuffer). In the example above, we're creating an "audio/mpeg" one, which is able to parse and decode our MP3 segments; there are several [other types](http://www.w3.org/2013/12/byte-stream-format-registry/) available.
+
+## Anomalous Waveforms
+
+We'll come back to the code in a moment, but lets now look more closely at the file we've just appended, specifically at the end of it. Below, is a graph of the last 3000 samples averaged across both channels from the _[sintel_0.mp3](sintel/sintel_0.mp3)_ track. Each pixel on the red line is a [floating point sample](http://en.wikipedia.org/wiki/Audio_bit_depth) in the range of [-1.0, 1.0].
+
+
+<p style="text-align: center;">
+  <img src="/web/updates/images/2015-06-12-media-source-extensions-for-audio/mp3_gap_end.png" alt="End of sintel_0.mp3" style="max-width: 60%; height: auto;">
+</p>
+
+
+What's with all that those zero (silent) samples!? They're actually due to [compression artifacts](http://en.wikipedia.org/wiki/Gapless_playback#Compression_artifacts) introduced during encoding. Almost every encoder introduces some type of padding. In this case [LAME](http://lame.sourceforge.net/) added exactly 576 padding samples to the end of the file.
+
+In addition to the padding at the end, each file also had padding added to the beginning. If we peek ahead at the _[sintel_1.mp3](sintel/sintel_1.mp3)_ track we'll see another 576 samples of padding exists at the front. The amount of padding varies by encoder and content, but we know the exact values based on [metadata](#parsing-gapless-metadata) included within each file.
+
+<div class="waveform-container">
+
+
+<p style="text-align: center;">
+  <img src="/web/updates/images/2015-06-12-media-source-extensions-for-audio/mp3_gap.png" alt="Beginning of sintel_1.mp3" style="max-width: 60%; height: auto;">
+</p>
+
+
+The sections of silence at the beginning and end of each file are what causes the "glitches" between segments in the previous demo. To achieve gapless playback, we need to remove these sections of silence. Luckily, this is easily done with MediaSource! Below, we'll modify our _onAudioLoaded()_ method to use an [append window](https://dvcs.w3.org/hg/html-media/raw-file/tip/media-source/media-source.html#definitions) and a [timestamp offset](https://dvcs.w3.org/hg/html-media/raw-file/tip/media-source/media-source.html#widl-SourceBuffer-timestampOffset) to remove this silence.
+
+## Example Code
+
+{% highlight javascript %}
```

Comment:
I think it's OK – the longer parts are mostly extended comments, and I think the examples are well constructed. 

(Note that the article is ported over from Dale's original [here](http://dalecurtis.github.io/llama-demo/index.html).)

If you feel strongly, we could ask Dale to shorten.